oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/osh/OSHEntityTimeUtils.java

```diff
@@ -0,0 +1,357 @@
+package org.heigit.ohsome.oshdb.util.osh;
+
+import static java.lang.String.format;
+
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.heigit.ohsome.oshdb.osh.OSHEntity;
+import org.heigit.ohsome.oshdb.osh.OSHNode;
+import org.heigit.ohsome.oshdb.osh.OSHRelation;
+import org.heigit.ohsome.oshdb.osh.OSHWay;
+import org.heigit.ohsome.oshdb.osm.OSMEntity;
+import org.heigit.ohsome.oshdb.osm.OSMMember;
+import org.heigit.ohsome.oshdb.osm.OSMNode;
+import org.heigit.ohsome.oshdb.osm.OSMRelation;
+import org.heigit.ohsome.oshdb.osm.OSMWay;
+import org.heigit.ohsome.oshdb.util.OSHDBTimestamp;
+
+public class OSHEntityTimeUtils {
+  private OSHEntityTimeUtils() {
+    throw new IllegalStateException("utility class");
+  }
+
+  /**
+   * Returns the changeset ids which correspond to modifications of this entity.
+   *
+   * <p>Used internally to group modifications by changeset.</p>
+   *
+   * @param osh the osh entity to work on
+   * @return a map between timestamps and changeset ids
+   */
+  public static Map<OSHDBTimestamp, Long> getChangesetTimestamps(OSHEntity osh) {
+    switch (osh.getType()) {
+      case NODE:
+        return getChangesetTimestamps((OSHNode) osh);
+      case WAY:
+        return getChangesetTimestamps((OSHWay) osh);
+      case RELATION:
+        return getChangesetTimestamps((OSHRelation) osh);
+      default:
+        throw new UnsupportedOperationException(format(UNSUPPORTED_OSMTYPE_MESSAGE, osh.getType()));
+    }
+  }
+
+  private static Map<OSHDBTimestamp, Long> getChangesetTimestamps(OSHNode osh) {
+    Map<OSHDBTimestamp, Long> result = new TreeMap<>();
+    osh.getVersions().forEach(osm -> result.put(osm.getTimestamp(), osm.getChangesetId()));
+    return result;
+  }
+
+  private static Map<OSHDBTimestamp, Long> getChangesetTimestamps(OSHWay osh) {
+    Map<OSHDBTimestamp, Long> result = new TreeMap<>();
+
+    List<OSMWay> ways = org.heigit.ohsome.oshdb.osh.OSHEntities.toList(osh.getVersions());
```

Comment:
Why do you use the full-qualified name for OSHEntities, instead of an import?

Reply:
Huh. It might be an artifact of the semi-automated refactoring done by my IDE when moving the [method](https://github.com/GIScience/oshdb/pull/332/files#diff-ee965dbc15238b368d4cbd6a2377ac995b40270c97a077fb5b64d21fff9d602aL134) to this new class. Thanks for catching this, I'll change it to an import. :+1: